src/libs/SessionUtils.js

```diff
@@ -0,0 +1,29 @@
+import lodashGet from 'lodash/get';
+
+/**
+ * Determine if the transitioning user is logging in as a new user.
+ *
+ * @param {String} transitionURL
+ * @param {String} sessionEmail
+ * @returns {Boolean}
+ */
+function isLoggingInAsNewUser(transitionURL, sessionEmail) {
+    // The OldDot mobile app does not URL encode the parameters, but OldDot web
+    // does. We don't want to deploy OldDot mobile again, so as a work around we
+    // compare both the raw and decoded email from the transition link. If both don't match
+    // the session email, then we are logging in as a new user.
+    // See https://github.com/Expensify/Mobile-Expensify/pull/12280#issuecomment-1155633503
+    const params = new URLSearchParams(transitionURL);
+    const paramsEmail = params.get('email');
+
+    // Capture the un-encoded text in the email param
+    const emailParamRegex = /[?&]email=([^&]*)/g;
+    const matches = emailParamRegex.exec(transitionURL);
+    const linkedEmail = lodashGet(matches, 1, null);
+    return paramsEmail !== sessionEmail && linkedEmail !== sessionEmail;
+}
+
+export {
+    // eslint-disable-next-line import/prefer-default-export
```

Comment:
Why not use default here?

Reply:
Which ever is convention here makes sense, it's just weird we have a convention that goes against our lint rules, right?

Reply:
I think our lint rules are too picky here. I'm going to stick with what I have and in the future when we can either update the lint rules or it's likely that we will add another session util.